python/ray/rllib/ddpg/ddpg_evaluator.py

```diff
@@ -0,0 +1,170 @@
+# imports
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import ray
+from ray.rllib.ddpg.models import DDPGModel
+from ray.rllib.ddpg.replay_buffer import ReplayBuffer, PrioritizedReplayBuffer
+from ray.rllib.models.catalog import ModelCatalog
+from ray.rllib.optimizers import SampleBatch
+from ray.rllib.optimizers import Evaluator
+from ray.rllib.utils.filter import NoFilter
+from ray.rllib.utils.process_rollout import process_rollout
+from ray.rllib.utils.sampler import SyncSampler
+from ray.tune.registry import get_registry
+
+import tensorflow as tf
+
+
+class DDPGEvaluator(Evaluator):
+
+
+    def __init__(self, registry, env_creator, config):
+        self.registry = registry
+        self.env = ModelCatalog.get_preprocessor_as_wrapper(
+            registry, env_creator(config["env_config"]), config["actor_model"])
+        self.config = config
+
+        self.sess = tf.Session()
+
+        self.model = DDPGModel(self.registry, self.env, self.config, self.sess)
+        self.target_model = DDPGModel(self.registry, self.env, self.config, self.sess)
+
+        self.initialize()
+        self.setup_gradients()
+        self._setup_target_updates()
+
+        self.replay_buffer = ReplayBuffer(config["buffer_size"])
+        self.sampler = SyncSampler(
+                        self.env, self.model, NoFilter(),
+                        config["batch_size"], horizon=config["horizon"])
+
+    def initialize(self):
+        self.sess.run(tf.global_variables_initializer())
+
+    #TODO: (not critical) Add batch normalization?
+    def sample(self, no_replay = True):
```

Comment:
Is it possible to remove the replay handling from DDPG, and use SyncLocalReplayOptimizer / ApexOptimizer instead? Recently we refactored DQN to have replay be handled by the optimizer classes.